R/proportional_hazards.R

```diff
@@ -206,6 +206,65 @@ predict_survival._coxnet <- function(object, new_data, ...) {
   predict_survival.model_fit(object, new_data = new_data, ...)
 }
 
+#' @export
+predict_raw._coxnet <- function(object, new_data, opts = list(), ...)  {
+  if (any(names(enquos(...)) == "newdata"))
+    rlang::abort("Did you mean to use `new_data` instead of `newdata`?")
+
+  object$spec <- eval_args(object$spec)
+  opts$s <- object$spec$args$penalty
+  predict_raw.model_fit(object, new_data = new_data, opts = opts, ...)
+}
+
+#' @export
+multi_predict._coxnet <- function(object,
+                                  new_data,
+                                  type = NULL,
+                                  penalty = NULL,
+                                  ...) {
+    if (any(names(enquos(...)) == "newdata"))
+      rlang::abort("Did you mean to use `new_data` instead of `newdata`?")
+
+    dots <- list(...)
+
+    object$spec <- eval_args(object$spec)
+
+    if (is.null(penalty)) {
+      # See discussion in https://github.com/tidymodels/parsnip/issues/195
+      if (!is.null(object$spec$args$penalty)) {
+        penalty <- object$spec$args$penalty
+      } else {
+        penalty <- object$fit$lambda
+      }
+    }
+
+    if (type != "linear_pred"){
+      pred <- predict(object, new_data = new_data, type = type, ...,
+                      penalty = penalty, multi = TRUE)
+    } else {
+      pred <- predict(object, new_data = new_data, type = "raw",
+                      opts = dots, penalty = penalty, multi = TRUE)
+
+      # post-processing into nested tibble
```

Comment:
There are currently predict methods for types `"linear_pred"` and `"survival"`. For the `"linear_pred"`-type predictions, this follows what parsnip does for linear_reg() with a glmnet engine. For the survival probabilities, we use the survival curves from `survfit()` and have the wrapper `survival_prob_coxnet()` already so I extended that one to be able to deal with a vector of penalties. This also allows for convenient minimal nesting where we only group according to strata, see also #47 and #63.